delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -1,16 +1,426 @@
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileState.{Current, Initial}
-import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.{FileCommand, FileEvent, FileRejection, FileState}
-import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils
-import monix.bio.{IO, UIO}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageRejection.{InvalidStorageId, StorageIsDeprecated}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.{DigestAlgorithm, Storage}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.{StorageResource, Storages}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.utils.EventLogUtils
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser.generateId
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label, ResourceRef, TagLabel}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.{IOUtils, UUIDF}
+import ch.epfl.bluebrain.nexus.delta.sdk.{Acls, AkkaSource, Organizations, Projects}
+import ch.epfl.bluebrain.nexus.sourcing._
+import ch.epfl.bluebrain.nexus.sourcing.processor.EventSourceProcessor.persistenceId
+import ch.epfl.bluebrain.nexus.sourcing.processor.ShardedAggregate
+import fs2.Stream
+import monix.bio.{IO, Task, UIO}
+import monix.execution.Scheduler
 
-class Files {}
+import java.util.UUID
+
+/**
+  * Operations for handling files
+  */
+final class Files(
+    formDataExtractor: FormDataExtractor,
+    aggregate: FilesAggregate,
+    eventLog: EventLog[Envelope[FileEvent]],
+    acls: Acls,
+    orgs: Organizations,
+    projects: Projects,
+    storages: Storages
+)(implicit uuidF: UUIDF, system: ClassicActorSystem, sc: Scheduler) {
+
+  // format: off
+  private val testStorageRef = ResourceRef.Revision(iri"http://localhost/test?rev=1", iri"http://localhost/test", 1)
+  private val testAttributes = FileAttributes(UUID.randomUUID(), "http://localhost", Uri.Path.Empty, "", `application/octet-stream`, 0, ComputedDigest(DigestAlgorithm.default, "value"))
+  // format: on
+
+  /**
+    * Create a new file where the id is  self generated
+    *
+    * @param storageId  the optional storage identifier to expand as the id of the storage. When None, the default storage is used
+    * @param projectRef the project where the file will belong
+    * @param entity     the http FormData entity
+    */
+  def create(
```

Comment:
We should trigger the digest computation in the end no ? 
Is this in another PR ?

Reply:
The digest computation is handled by the `storage.saveFile`